tests/test_graph6.py

```diff
@@ -0,0 +1,262 @@
+
+            # check nodes and edges count
+            self.assertEqual(g2.num_nodes(), 2)
+            self.assertEqual(g2.num_edges(), 1)
+
+            # graph6 doesn't guarantee attributes; allow None or preserved dict
+            n0 = g2[0]
+            self.assertTrue(n0 is None or ("label" in n0 and n0["label"] == "n0"))
+
+            # check edge exists
+            self.assertTrue(list(g2.edge_list()))
+        finally:
+            os.remove(path)
+
+    def test_read_graph6_str_undirected(self):
+        """Test reading an undirected graph from a graph6 string."""
+        g6_str = "A_"
+        graph = rx_graph6.read_graph6_str(g6_str)
+        self.assertIsInstance(graph, rx.PyGraph)
+        self.assertEqual(graph.num_nodes(), 2)
+        self.assertEqual(graph.num_edges(), 1)
+        self.assertTrue(graph.has_edge(0, 1))
+
+    def test_read_graph6_str_directed(self):
+        """Test reading a directed graph from a graph6 string."""
+        g6_str = "&AG"
+        graph = rx_digraph6.read_graph6_str(g6_str)
+        self.assertIsInstance(graph, rx.PyDiGraph)
+        self.assertEqual(graph.num_nodes(), 2)
+        self.assertEqual(graph.num_edges(), 1)
+        self.assertTrue(graph.has_edge(1, 0))
+
+    def test_write_graph6_from_pygraph(self):
+        """Test writing a PyGraph to a graph6 string."""
+        graph = rx.PyGraph()
+        graph.add_nodes_from(range(2))
+        graph.add_edge(0, 1, None)
+        g6_str = rx_graph6.write_graph6_from_pygraph(graph)
+        self.assertEqual(g6_str, "A_")
+
+    def test_write_graph6_from_pydigraph(self):
+        """Test writing a PyDiGraph to a graph6 string."""
+        graph = rx.PyDiGraph()
+        graph.add_nodes_from(range(2))
+        graph.add_edge(1, 0, None)
+        g6_str = rx_digraph6.write_graph6_from_pydigraph(graph)
+        self.assertEqual(g6_str, "&AG")
+
+    def test_roundtrip_undirected(self):
+        """Test roundtrip for an undirected graph."""
+        graph = rx.generators.path_graph(4)
+        g6_str = rx_graph6.write_graph6_from_pygraph(graph)
+        new_graph = rx_graph6.read_graph6_str(g6_str)
+        self.assertEqual(graph.num_nodes(), new_graph.num_nodes())
+        self.assertEqual(graph.num_edges(), new_graph.num_edges())
+        self.assertEqual(graph.edge_list(), new_graph.edge_list())
+
+    def test_roundtrip_directed(self):
+        """Test roundtrip for a directed graph."""
+        graph = rx.generators.directed_path_graph(4)
+        g6_str = rx_digraph6.write_graph6_from_pydigraph(graph)
+        new_graph = rx_digraph6.read_graph6_str(g6_str)
+        self.assertEqual(graph.num_nodes(), new_graph.num_nodes())
+        self.assertEqual(graph.num_edges(), new_graph.num_edges())
+        self.assertEqual(graph.edge_list(), new_graph.edge_list())
+
+    def test_read_graph6_file(self):
+        """Test reading a graph from a graph6 file."""
+        with tempfile.NamedTemporaryFile(mode="w", delete=False) as fd:
```

Comment:
I don't see why the whole test couldn't exist within the `with` statement. I'd rather not have a `os.remove` statement and let `tempfile` deal with it

Example: https://github.com/Qiskit/rustworkx/blob/7318a803dc87853cbb2e916d7095b440985507e0/tests/digraph/test_node_link_json.py#L116